src/lib/storage/patches.rb

```diff
@@ -63,12 +63,97 @@ def dasd_type
     def dasd_format
       ::Storage::DASDF_NONE
     end
+
+    def size_k=(s)
+      self.size = s * 1024
+    end
+
+    def size_k
+      self.size / 1024
+    end
+
   end
 
   # patch libstorage-ng class
   class Partition
     def inspect
-      "<Partition \##{sid} #{name} #{Yast::Storage::DiskSize.KiB(size_k)}>"
+      "<Partition #{name} #{Yast::Storage::DiskSize.KiB(size_k)}, #{region.show_range}>"
+    end
+
+    def size_k=(s)
+      self.size = s * 1024
+    end
+
+    def size_k
+      self.size / 1024
+    end
+
+  end
+
+  class PartitionSlot
+    def inspect
+      flags = ""
+      flags += "P" if self.primary_slot
+      flags += "p" if self.primary_possible
+      flags += "E" if self.extended_slot
+      flags += "e" if self.extended_possible
+      flags += "L" if self.logical_slot
+      flags += "l" if self.logical_possible
+      nice_size = Yast::Storage::DiskSize.B(region.length * region.block_size)
+      "<PartitionSlot #{self.nr} #{self.name} #{flags} #{nice_size}, #{self.region.show_range}>"
+    end
+
+    alias to_s inspect
+
+    def type?(type)
```

Comment:
It is planned to add these methods to libstorage-ng any time soon? That's the goal of `patches.rb`, to temporarily provide stuff that will be eventually provided by the library.